metricq/exceptions.py

```diff
@@ -0,0 +1,55 @@
+"""
+This module defines custom Exception/Error classes for MetricQ.
+In general, they are used for errors in the interaction with other agents and subclassed by specific built-in errors rather than one "MetricqError"
+Misuse of the API generally causes built-in errors, i.e. TypeError, ValueError, and, KeyError.
+AssertionError / assert is only used for checking invariants within the library itself.
+"""
+
+
+class RPCError(Exception):
+    """ An RPC returned an error code. This could be an issue with the input or on the other side. """
+
+    pass
+
+
+class MessageError(Exception):
+    """ An error for something that is semantically wrong within a protobuf or RPC message """
+
+    pass
+
+
+class HistoryResponseError(MessageError):
+    def __init__(self, extra_message):
+        super().__init__(
+            (
+                "Failed to parse inconsistent history response message: {}."
```

Comment:
This message is too specific for that error. Either I dump the error message from the DB into this and it gets confusing, or I define my own Error class.

Reply:
Also, there still is the `InvalidHistoryResponse` in history_client.py. They seem to do the same stuff somehow.